torax/_src/neoclassical/transport/base.py

```diff
@@ -64,22 +66,113 @@ def zeros(cls, geometry: geometry_lib.Geometry) -> 'NeoclassicalTransport':
 class NeoclassicalTransportModel(abc.ABC):
   """Base class for neoclassical transport models."""
 
-  @abc.abstractmethod
-  def calculate_neoclassical_transport(
+  def __call__(
       self,
       runtime_params: runtime_params_slice.RuntimeParams,
       geometry: geometry_lib.Geometry,
       core_profiles: state.CoreProfiles,
   ) -> NeoclassicalTransport:
     """Calculates neoclassical transport."""
+    neoclassical_transport = self._call_implementation(
+        runtime_params,
+        geometry,
+        core_profiles,
+    )
+    neoclassical_transport = self._apply_clipping(
+        runtime_params,
+        neoclassical_transport,
+    )
+    return neoclassical_transport
+
+  def _apply_clipping(
+      self,
+      runtime_params: runtime_params_slice.RuntimeParams,
+      neoclassical_transport: NeoclassicalTransport,
+  ) -> NeoclassicalTransport:
+    chi_neo_i = jnp.clip(
+        neoclassical_transport.chi_neo_i,
+        runtime_params.neoclassical.transport.chi_min,
+        runtime_params.neoclassical.transport.chi_max,
+    )
+    chi_neo_e = jnp.clip(
+        neoclassical_transport.chi_neo_e,
+        runtime_params.neoclassical.transport.chi_min,
+        runtime_params.neoclassical.transport.chi_max,
+    )
+    D_neo_e = jnp.clip(
+        neoclassical_transport.D_neo_e,
+        runtime_params.neoclassical.transport.D_e_min,
+        runtime_params.neoclassical.transport.D_e_max,
+    )
+    V_neo_e = jnp.clip(
+        neoclassical_transport.V_neo_e,
+        runtime_params.neoclassical.transport.V_e_min,
+        runtime_params.neoclassical.transport.V_e_max,
+    )
+    V_neo_ware_e = jnp.clip(
+        neoclassical_transport.V_neo_ware_e,
+        runtime_params.neoclassical.transport.V_e_min,
+        runtime_params.neoclassical.transport.V_e_max,
+    )
+    return dataclasses.replace(
```

Comment:
You are updating all  the fields here. Why not just create a new object to make that clear